src/urllib3/response.py

```diff
@@ -148,6 +153,24 @@ def flush(self) -> bytes:
             return b""
 
 
+if zstd is not None:
+
+    class ZstdDecoder(ContentDecoder):
+        def __init__(self) -> None:
+            self._obj = zstd.EndlessZstdDecompressor()
+
+        def decompress(self, data: bytes) -> bytes:
+            return self._obj.decompress(data) if data else data  # type: ignore[no-any-return]
```

Comment:
This line seems more complex than it needs to be, maybe we can extract the empty case into it's own block for better readability?